### PR TITLE
added explicit return type definitions

### PR DESCRIPTION
**Pull request by KyleCartmell**
added explicit return type definitions to life and ofType member functions within ActionsObservable class (github issue #96)

### CODE REVIEW

**Comment by KyleCartmell** [comment]
I am unable to reproduce this build failure locally (my local build works perfectly with and without the changes in this PR) and the error output from the log isn't great. I'd be interested to see if the issue is repeatable but I don't see a way to re-run the build. Can someone take a look at this for me?

**Comment by KyleCartmell** [comment]
Ahh, there is a problem here. The return type on ofType definitely shouldn't be boolean, it's an observable of some sort. I'm puzzling through precisely what types would be best exposed here.

Not sure that this problem is the cause of the code 2 return from tsc in the CI build; I get some awesome error messages out of the test harness now that I'm invoking it correctly.

I'll close this and try to do something smarter, but feedback would still be greatly appreciated.

File: index.d.ts [hunk 1]
```diff
@@ -4,8 +4,8 @@ import { Operator } from 'rxjs/Operator';
 
 export declare class ActionsObservable extends Observable<Action> {
   constructor(actionsSubject: Observable<Action>);
-  lift(operator: Operator<any, Action>);
-  ofType(...key: any[]);
+  lift(operator: Operator<any, Action>) : ActionsObservable;
+  ofType(...key: any[]) : boolean;
```
Comment (gusvargas) [inline index.d.ts]: https://travis-ci.org/redux-observable/redux-observable/builds/152735720#L388

```
test/typings.ts(10,6): error TS2339: Property 'mapTo' does not exist on type 'boolean'.
```

`ofType` also returns `ActionsObservable`, not `boolean`. The boolean returns you see in [`ofType`](https://github.com/redux-observable/redux-observable/blob/master/src/ActionsObservable.js#L16-L30) are for returning from the `filter` predicate.

Comment (gusvargas) [inline index.d.ts]: Oops. You beat me to finding the problem. 👍

Comment (KyleCartmell) [inline index.d.ts]: I think that lift should explicitly return ActionsObservable but I'm not sure about ofType.
(full disclosure: I'm still trying to grok RxJS.)

ofType does a filter thing, which returns an Operator, but Operator is Observable, right? So... explicitly returning Operator doesn't seem wrong but I'm not sure that's semantically correct.

Comment (jayphelps) [inline index.d.ts]: Both `lift` and `ofType` return `ActionsObservable`, though indeed that's not currently annotated.

**Comment by jayphelps** [comment]
@KyleCartmell reopened based on our discussion in #96. Please amend your existing commit with the fix--it also needs to follow our [commit message formatting requirements](https://github.com/redux-observable/redux-observable/blob/master/CONTRIBUTING.md)

**Comment by KyleCartmell** [comment]
@jayphelps corrected for review

Thanks for the help and lesson!

**Comment by jayphelps** [comment]
LGTM. Thanks @KyleCartmell !

**Comment by jayphelps** [comment]
@KyleCartmell noticed you cut your own build so wanted to let you know that 0.9.1 was released and includes this fix. 🍻

**Comment by KyleCartmell** [comment]
Sadly, I'm stuck on 0.8.0 until https://github.com/angular/angular/issues/9549.

For anyone else trying to follow this thread later, I discovered after I'd pulled the declaration into the 0.8.0 version that I could have (and now have) just added the the declaration to the typings mechanism in angular2-seed which feeds into tsc and does the needful. (I'm also a version behind on angular2-seed (2.0.0-rc4) so good luck, future humans, if you're trying to do this thing!)

**Pull request merged**